libraries/AP_Mount/AP_Mount_CADDX.h

```diff
@@ -0,0 +1,55 @@
+/*
+  CADDX mount using serial protocol backend class
+ */
+#pragma once
+
+#include "AP_Mount_config.h"
+
+#if HAL_MOUNT_CADDX_ENABLED
+
+#include "AP_Mount_Backend_Serial.h"
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Math/AP_Math.h>
+#include <AP_Common/AP_Common.h>
+
+class AP_Mount_CADDX : public AP_Mount_Backend_Serial
+{
+
+public:
+    // Constructor
+    using AP_Mount_Backend_Serial::AP_Mount_Backend_Serial;
+
+    // update mount position - should be called periodically
+    void update() override;
+
+    // has_pan_control - returns true if this mount can control its pan (required for multicopters)
+    bool has_pan_control() const override { return yaw_range_valid(); };
+
+    // has_roll_control - returns true if this mount can control its roll
+    bool has_roll_control() const override { return roll_range_valid(); };
+
+    // has_pitch_control - returns true if this mount can control its tilt
+    bool has_pitch_control() const override { return pitch_range_valid(); };
+
+protected:
+
+    // get attitude as a quaternion.  returns true on success
+    bool get_attitude_quaternion(Quaternion& att_quat) override;
```

Comment:
```suggestion
    bool get_attitude_quaternion(class Quaternion& att_quat) override;
```

Not actually including the right header to get this - just getting it transitively somehow

Reply:
I'm going to skip this change I think because it leads to a compile failure and would make it inconsistent with the virtual method that it is overriding and also all other drivers.  Apologies, I'm probably misunderstanding something